apps/system/sekey_test/sekey_test.c

```diff
@@ -0,0 +1,243 @@
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <tinyara/security_hal.h>
+#include <tinyara/seclink.h>
+#include <mbedtls/x509_crt.h>
+
+#ifdef CONFIG_ECDSA_KEY_SLOT
+#define ECDSA_KEY_SLOT CONFIG_ECDSA_KEY_SLOT
+#else
+#define ECDSA_KEY_SLOT "0"
+#endif
+
+#ifdef CONFIG_CERT_SLOT
+#define CERT_SLOT CONFIG_CERT_SLOT
+#else
+#define CERT_SLOT "1,2,3"
+#endif
+
+#define HASH_LEN 32
+#define SIGN_LEN 32
+#define CERT_MAX_SIZE 1024
+
+static uint8_t key_test[128] = {
+	0,
+};
+static uint8_t cert_test[128] = {
+	0,
+};
+
+int sekey_test_main(int argc, char *argv[])
+{
+	int i;
+	int ret;
+	int res = 0;
+	char *ptr;
+	sl_ctx shnd;
+	hal_result_e hres = 0;
+	hal_ecdsa_mode ecdsa_mode;
+	hal_data t_hash = {
+		0,
+	};
+	hal_data sign = {
+		0,
+	};
+
+	mbedtls_x509_crt cert = {
+		0,
+	};
+	hal_data cert_out = {
+		0,
+	};
+	char buf[256] = {
+		0,
+	};
+
+	char key_temp[] = ECDSA_KEY_SLOT;
+	char cert_temp[] = CERT_SLOT;
+	int key_cnt = 0;
+	int cert_cnt = 0;
+
+	ret = sl_init(&shnd);
+	if (ret != SECLINK_OK) {
+		printf("[SECLINK] sl_init fail \n");
+		return -1;
+	}
+
+	ptr = strtok(key_temp, ",");
+	while (ptr != NULL) {
+		key_test[key_cnt++] = atoi(ptr);
```

Comment:
I think you need to check `key_cnt` and `cert_cnt` values with 128(buffer size).
That's why it can make an overflow here if user set `CONFIG_CERT_SLOT` or `CONFIG_ECDSA_KEY_SLOT` as a long list with larger than buffer size.
Please add checking them with a range of buffer.
And you'd better to define values like `#define KEY_CNT 128` and `#define CERT_CNT 128` and use them when checking the range for readability and security.

Reply:
@jeongarmy 
Good point.  I will fix my code